src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -512,39 +509,51 @@ private static bool AreEqualNetworkCredentials(NetworkCredential credential1, Ne
         {
-            // If we weren't handed a multi agent, see if we can get one from the EasyRequest
-            if (agent == null && easy != null && easy._associatedMultiAgent != null)
+            if (EventSourceTracingEnabled)
             {
-                agent = easy._associatedMultiAgent;
+                EventSourceTraceCore(string.Format(formatMessage, arg0), agent, easy, memberName);
             }
-            int? agentId = agent != null ? agent.RunningWorkerId : null;
+        }
 
-            // Get an ID string that provides info about which MultiAgent worker and which EasyRequest this trace is about
-            string ids = "";
-            if (agentId != null || easy != null)
+        private static void EventSourceTrace<TArg0, TArg1, TArg2>
+            (string formatMessage, TArg0 arg0, TArg1 arg1, TArg2 arg2,
+            MultiAgent agent = null, EasyRequest easy = null, [CallerMemberName] string memberName = null)
+        {
+            if (EventSourceTracingEnabled)
             {
-                ids = "(" +
-                    (agentId != null ? "M#" + agentId : "") +
-                    (agentId != null && easy != null ? ", " : "") +
-                    (easy != null ? "E#" + easy.Task.Id : "") +
-                    ")";
+                EventSourceTraceCore(string.Format(formatMessage, arg0, arg1, arg2), agent, easy, memberName);
             }
+        }
 
-            // Create the message and trace it out
-            string msg = string.Format("[{0, -30}]{1, -16}: {2}", memberName, ids, text);
-            Interop.Sys.PrintF("%s\n", msg);
+        private static void EventSourceTrace(
+            string message = null, 
+            MultiAgent agent = null, EasyRequest easy = null, [CallerMemberName] string memberName = null)
+        {
+            if (EventSourceTracingEnabled)
+            {
+                EventSourceTraceCore(message, agent, easy, memberName);
+            }
         }
 
-        [Conditional(VerboseDebuggingConditional)]
-        private static void VerboseTraceIf(bool condition, string text = null, [CallerMemberName] string memberName = null, EasyRequest easy = null)
+        private static void EventSourceTraceCore(string message, MultiAgent agent, EasyRequest easy, string memberName)
         {
-            if (condition)
+            // If we weren't handed a multi agent, see if we can get one from the EasyRequest
+            if (agent == null && easy != null && easy._associatedMultiAgent != null)
             {
-                VerboseTrace(text, memberName, easy, agent: null);
+                agent = easy._associatedMultiAgent;
             }
+
+            HttpEventSource.Log.HandlerMessage(
+                agent != null && agent.RunningWorkerId.HasValue ? agent.RunningWorkerId.GetValueOrDefault() : 0,
+                easy != null ? easy.Task.Id : 0,
+                memberName ?? string.Empty,
```

Comment:
This is the only call site.  But, yeah, it's possible there will be others in the future.  I'll move the checks.